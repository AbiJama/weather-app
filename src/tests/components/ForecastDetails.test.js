// ForecastDetails.test.js

import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe('ForecastDetails component', () => {
    const validProps = {
      date: 1111111,
      temperature: {
        max: 50,
        min: 3,
      },
      humidity: 12,
      wind: {
        speed: 99,
        direction: 'n',
      },
    };
  
    it('renders correctly', () => {
      const { asFragment } = render(<ForecastDetails forecast={validProps} />);
  
      expect(asFragment()).toMatchSnapshot();
    });
  });