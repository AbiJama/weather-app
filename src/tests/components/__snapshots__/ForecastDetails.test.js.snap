// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date "
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__min-temperature"
    >
      Min Temperature: 3°C
    </div>
    <div
      class="forecast-details__max-temperature"
    >
      Max Temperature: 50°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 12%
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      Wind Speed: 99 mph
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      Wind Direction: n
    </div>
  </div>
</DocumentFragment>
`;
